python/cog/server/worker.py

```diff
@@ -155,20 +181,40 @@ def _setup(self) -> None:
             # otherwise, if setup is sync and the user does new_event_loop to use a ClientSession,
             # then tries to use the same session from async predict, they would get an error.
             # that's significant if connections are open and would need to be discarded
+            debug("async predictor")
             if is_async_predictor(self._predictor):
+                debug("getting loop")
                 self.loop = get_loop()
+                debug("got loop")
+            debug("getattr")
             # Could be a function or a class
             if hasattr(self._predictor, "setup"):
+                debug("inspect")
                 if inspect.iscoroutinefunction(self._predictor.setup):
-                    self.loop.run_until_complete(run_setup_async(self._predictor))
+
+                    async def setup_async() -> None:
+                        # we prefer to not stop-start the event loop between these calls
+                        await self._async_init()  # this creates tasks
+                        debug("async_init done")
+                        await run_setup_async(self._predictor)
+                        debug("ran setup async")
```

Comment:
not sure if this is still needed

Reply:
Repeating back to confirm my own understanding: We no longer need `setup_async`, because our new approach is to always run (sync) `setup` and then transition to async at the end after we know the signature of `predict`. Right?

Reply:
no, setup may be async -- what I mean here is that 
```suggestion
                    self.loop.run_until_complete(self._async_init())
                    self.loop.run_until_complete(run_setup_async(self._predictor))
```
I added the setup_async function there so that the event loop kept going between the two coros because I thought that was causing problems, but I'm not sure that's necessary